File: 0101-0200/0119.Pascals_Triangle_II.cpp
//0ms


class Solution {
public:
    std::vector<int> getRow(int rowIndex) {
        std::vector<int> res(1, 1);
        long long prev = 1;
        for(int k = 1; k <= rowIndex; k++) {
            long long next_val = prev * (rowIndex - k + 1) / k;
            res.push_back(next_val);
            prev = next_val;
        }
        return res;
    }
};

File: 0101-0200/0119.Pascals_Triangle_II.rs
//0ms


impl Solution {
    pub fn get_row(row_index: i32) -> Vec<i32> {
        let k = row_index as usize;
        let mut row = vec![1; k + 1];             
        for i in 2..=k {                          
            for j in (1..i).rev() {            
                row[j] = row[j - 1] + row[j];     
            }
        }
        row
    }
}
